aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -1485,15 +1500,23 @@ func resourceAviatrixSpokeGatewayUpdate(d *schema.ResourceData, meta interface{}
 		haEip := d.Get("ha_eip").(string)
 		if goaviatrix.IsCloudType(spokeGw.CloudType, goaviatrix.AWSRelatedCloudTypes|goaviatrix.GCPRelatedCloudTypes|goaviatrix.OCIRelatedCloudTypes) {
 			spokeGw.Eip = haEip
-		} else if goaviatrix.IsCloudType(spokeGw.CloudType, goaviatrix.AzureArmRelatedCloudTypes) && haEip != "" && spokeGw.GwSize != "" {
-			// No change will be detected when ha_eip is set to the empty string because it is computed.
-			// Instead, check ha_gw_size to detect when HA gateway is being deleted.
-			haAzureEipName, ok := d.GetOk("ha_azure_eip_name_resource_group")
-			if !ok {
-				return fmt.Errorf("failed to create HA Spoke Gateway: 'ha_azure_eip_name_resource_group' must be set when a custom EIP is provided and cloud_type is Azure (8), AzureGov (32) or AzureChina (2048)")
-			}
-			// AVX-9874 Azure EIP has a different format e.g. 'test_ip:rg:104.45.186.20'
-			spokeGw.Eip = fmt.Sprintf("%s:%s", haAzureEipName.(string), haEip)
+		}
+
+		haAzureEipName, haAzureEipNameOk := d.GetOk("ha_azure_eip_name_resource_group")
+		if goaviatrix.IsCloudType(spokeGw.CloudType, goaviatrix.AzureArmRelatedCloudTypes) {
+			if haEip != "" && spokeGw.GwSize != "" {
+				// No change will be detected when ha_eip is set to the empty string because it is computed.
+				// Instead, check ha_gw_size to detect when HA gateway is being deleted.
+				if !haAzureEipNameOk {
+					return fmt.Errorf("failed to create HA Spoke Gateway: 'ha_azure_eip_name_resource_group' must be set when a custom EIP is provided and cloud_type is Azure (8), AzureGov (32) or AzureChina (2048)")
+				}
+				// AVX-9874 Azure EIP has a different format e.g. 'test_ip:rg:104.45.186.20'
+				spokeGw.Eip = fmt.Sprintf("%s:%s", haAzureEipName.(string), haEip)
+			}
+		} else {
+			if haAzureEipNameOk {
```

Comment:
In some spots you do `else if` in others you do this with `else` and `if` on separate lines. Please pick one style and be consistent.